R/verb-copy-to.R

```diff
@@ -87,6 +87,64 @@ copy_to.src_sql <- function(dest, df, name = deparse(substitute(df)),
   invisible(out)
 }
 
+#' @export
+db_values <- function(con, df, name = deparse(substitute(df))) {
+  if (ncol(df) == 0) {
+    abort("`df` needs at least one column.")
+  }
+
+  if (!inherits(df, "data.frame")) {
+    abort("`df` needs to be a data.frame.")
+  }
+
+  tbl(
+    src = con,
+    from = sql_values(con, df),
+    vars = colnames(df)
+  )
+}
+
+sql_values <- function(con, df) {
+  UseMethod("sql_values")
+}
+
+#' @export
+sql_values.DBIConnection <- function(con, df) {
+  sql_values_clause(con, df)
+}
+
+sql_values_clause <- function(con, df, row = FALSE) {
+  sim_data <- rep_named(colnames(df), list(NULL))
+  cols_clause <- escape(sim_data, con = con, parens = FALSE)
+  null_row_clause <- build_sql("SELECT ", cols_clause, " WHERE false", con = con)
```

Comment:
```suggestion
  null_row_clause <- build_sql("SELECT ", cols_clause, " WHERE 0 = 1", con = con)
```

Not all databases know about logical literals.